### PR TITLE
Add copyable command for deployment

### LINKED ISSUE #2667: `sncast account create` should show copyable command for deployment
### Which components does the task require to be changed? (think hard pls)

sncast

### Description

Right now, when `sncast account create` is used, it is unclear how to deploy the created account.
The output already indicates that the account needs to be prefunded but doesn't show how to do so.

Change `sncast account create` output, so it also includes a ready to copy command for declaring. The arguments like `--url` that are provided by the user should be captured and included in the output command so it can be copied and run.

Concept:

```shell
$ sncast account create --url my_url.xyz --name my_account
add_profile: --add-profile flag was not set. No profile added to snfoundry.toml
address: 0x075c4094d5431fea3bdfa1c65d393ea52315f513b9fa81d1466edf1d91fc4022
max_fee: 4241681639235
message: Account successfully created. Prefund generated address with at least <max_fee> STRK tokens or an equivalent amount of ETH tokens. It is good to send more in the case of higher demand.

After prefunding the address run:
sncast account deploy --url my_url.xyz --name my_account --fee-token strk 
```

### PR DESCRIPTION
**Pull request by kkawula**
Closes #2667

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/src/starknet_commands/account/create.rs [hunk 1]
```diff
@@ -106,11 +109,25 @@ pub async fn create(
...
+                rpc_url,
+                account
+            )
+            .as_str(),
+        );
     }
 
     if add_profile.is_some() {
         let config = CastConfig {
-            url: create.rpc.url.clone().unwrap_or_default(),
```
Comment (kkawula) [inline crates/sncast/src/starknet_commands/account/create.rs:113]: It didn't work in case where url is provided from profile

Comment (kkawula) [reply crates/sncast/src/starknet_commands/account/create.rs:113]: rare but possible

File: crates/sncast/src/starknet_commands/account/create.rs [hunk 2]
```diff
@@ -106,11 +109,25 @@ pub async fn create(
...
+                if accounts_file.to_string().contains(DEFAULT_ACCOUNTS_FILE) {
+                    format!(" --accounts-file {accounts_file}")
+                } else {
+                    String::new()
+                },
+                rpc_url,
+                account
+            )
+            .as_str(),
+        );
```
Comment (franciszekjob) [inline crates/sncast/src/starknet_commands/account/create.rs:125]:
```suggestion
        let accounts_flag = if accounts_file.contains(DEFAULT_ACCOUNTS_FILE) {
            format!(" --accounts-file {}", accounts_file)
        } else {
            String::new()
        };

        let deploy_command = format!(
            "\n\nAfter prefunding the address, run:\n\
        sncast{} account deploy --url {} --name {} --fee-token strk",
            accounts_flag, rpc_url, account
        );

        message.push_str(&deploy_command);
```

**Review by franciszekjob — APPROVED** [review]

File: crates/sncast/src/starknet_commands/account/create.rs [hunk 3]
```diff
@@ -106,11 +109,14 @@ pub async fn create(
         )?;
     } else {
         write_account_to_accounts_file(account, accounts_file, chain_id, account_json.clone())?;
+
+        let deploy_command = generate_deploy_command(accounts_file, &rpc_url, account);
+        message.push_str(&deploy_command);
```
Comment (cptartur) [inline crates/sncast/src/starknet_commands/account/create.rs:114]: Why aren't we adding this if keystore is used? If the command would have to be different, I'd either resolve that or add a follow-up issue.

File: crates/sncast/src/starknet_commands/account/create.rs [hunk 4]
```diff
@@ -314,3 +320,16 @@ fn write_account_to_file(
     )?;
     Ok(())
 }
+
+fn generate_deploy_command(accounts_file: &Utf8PathBuf, rpc_url: &str, account: &str) -> String {
+    let accounts_flag = if accounts_file.to_string().contains(DEFAULT_ACCOUNTS_FILE) {
+        format!(" --accounts-file {accounts_file}")
```
Comment (cptartur) [inline crates/sncast/src/starknet_commands/account/create.rs:326]: Shouldn't this one be only displayed if non default accounts file is used?

Comment (cptartur) [reply crates/sncast/src/starknet_commands/account/create.rs:326]: + I'd handle it in a simpler manner: if user passed `--accounts-file` then show the command that uses it, if not, then do not show anything.

**Review by cptartur — CHANGES_REQUESTED** [review]

File: crates/sncast/src/starknet_commands/account/create.rs (hunk 4 above)
Comment (kkawula) [reply crates/sncast/src/starknet_commands/account/create.rs]: It looks like this:
without account file 
command:
```shell
sncast account create --name test1 -c 0x61dac032f228abef9c6626f995015233097ae253a7f72d68552db02f2971b8f --type oz --add-profile test1
```

output:
```shell
command: account create
add_profile: Profile test1 successfully added to snfoundry.toml
address: 0x0440ffd0fffbcdf12bc0ac683a0d1d7221336f5e67894346903deb23511722bb
max_fee: 30100000000000
message: Account successfully created. Prefund generated address with at least <max_fee> STRK tokens or an equivalent amount of ETH tokens. It is good to send more in the case of higher demand.

After prefunding the address, run:
sncast account deploy --url http://127.0.0.1:5050 --name test1 --fee-token strk

To see account creation details, visit:
account: https://sepolia.starkscan.co/contract/0x440ffd0fffbcdf12bc0ac683a0d1d7221336f5e67894346903deb23511722bb
```

with account file
command:
```shell
sncast --accounts-file konta.json account create --name test2 -c 0x61dac032f228abef9c6626f995015233097ae253a7f72d68552db02f2971b8f --type oz --add-profile test2
```

output:
```shell
command: account create
add_profile: Profile test2 successfully added to snfoundry.toml
address: 0x01d1efd96c297f2383ce5debf7353ad54852c2673d1b033f1a94f2e50a4f4b60
max_fee: 30100000000000
message: Account successfully created. Prefund generated address with at least <max_fee> STRK tokens or an equivalent amount of ETH tokens. It is good to send more in the case of higher demand.

After prefunding the address, run:
sncast --accounts-file konta.json account deploy --url http://127.0.0.1:5050 --name test2 --fee-token strk

To see account creation details, visit:
account: https://sepolia.starkscan.co/contract/0x1d1efd96c297f2383ce5debf7353ad54852c2673d1b033f1a94f2e50a4f4b60
```

File: crates/sncast/src/starknet_commands/account/create.rs (hunk 3 above)
Comment (kkawula) [reply crates/sncast/src/starknet_commands/account/create.rs:117]: Added

**Review by cptartur — APPROVED** [review]

**Pull request merged**